frontend/templates/product/hooks/useSelectedVariant.ts

```diff
@@ -1,70 +1,64 @@
 import {
-   invariant,
-   getVariantIdFromVariantURI,
    getProductVariantURI,
+   getVariantIdFromVariantURI,
 } from '@ifixit/helpers';
 import type { Product, ProductVariant } from '@pages/api/nextjs/cache/product';
-import { useRouter } from 'next/router';
-import { useCallback } from 'react';
+import { defaultVariantFor } from 'app/_helpers/product-helpers';
+import { useRouter, useSearchParams } from 'next/navigation';
 
-type SetVariantIdFn = (variantId: string) => void;
+type SelectVariantFn = (variantId: string) => void;
 
 export function useSelectedVariant(
    product: Product
-): [ProductVariant, SetVariantIdFn] {
+): [ProductVariant, SelectVariantFn] {
    const router = useRouter();
+   const searchParams = useSearchParams();
 
-   const defaultVariantId = useDefaultVariantId(product);
-   const searchParamVariantId = useSearchParamVariantId();
+   const selectVariant = (variantId: string) => {
+      router.replace(variantUrlFor(variantId), { scroll: false });
+   };
 
-   const currentVariantId = searchParamVariantId ?? defaultVariantId;
+   return [selectedVariant(), selectVariant];
 
-   let variant = product.variants.find((v) => v.id === currentVariantId);
-   if (variant == null) {
-      variant = product.variants.find((v) => v.id === defaultVariantId);
+   function selectedVariant() {
+      return searchParamVariant() ?? defaultVariant();
    }
 
-   invariant(
-      variant,
-      `Something went wrong, variant with id "${currentVariantId}" not found`
-   );
+   function searchParamVariant() {
+      if (variantGIDParam() == null) return null;
+      return product.variants.find((v) => v.id === variantGIDParam()) ?? null;
+   }
 
-   const setVariantId = useCallback<SetVariantIdFn>(
-      (variantId) => {
-         const { variant, ...newQuery } = router.query;
-         if (variantId !== defaultVariantId) {
-            newQuery.variant = getVariantIdFromVariantURI(variantId);
-         }
-         router.replace(
-            {
-               query: newQuery,
-            },
-            undefined,
-            { shallow: true }
-         );
-      },
-      [defaultVariantId, router]
-   );
+   function defaultVariant() {
+      return defaultVariantFor(product);
+   }
 
-   return [variant, setVariantId];
-}
+   function variantUrlFor(variantId: string) {
+      const newSearchParams = searchParamsCopy();
```

Comment:
```suggestion
      const newSearchParams = new URLSearchParams(searchParams.entries());
```

`.entries` produces pairs of values which is acceptable input to the constructor.